pwndbg/gdblib/arch.py

```diff
@@ -44,16 +44,18 @@
 
 def _get_arch(ptrsize: int):
     not_exactly_arch = False
+    endian_value = gdb.execute("show endian", to_string=True)
 
-    if "little" in gdb.execute("show endian", to_string=True).lower():
+    if endian_value and "little" in endian_value.lower():
         endian = "little"
     else:
         endian = "big"
 
     if pwndbg.gdblib.proc.alive:
         arch = gdb.newest_frame().architecture().name()
     else:
-        arch = gdb.execute("show architecture", to_string=True).strip()
+        arch_value = gdb.execute("show architecture", to_string=True)
+        arch = arch_value.strip() if arch_value else "UNKNOWN"
```

Comment:
we should assume the 'show arch' always work

Reply:
fixed!